lib/utils.pm

```diff
@@ -651,6 +668,10 @@ sub power_action {
     }
     else {
         assert_shutdown($shutdown_timeout) if $action eq 'poweroff';
+        # we must only reset consoles after we are really sure the system was
+        # rebooted otherwise we would look for a login prompt in the next
+        # select_console call while the consoles are still logged in
+        handle_livecd_reboot_failure if get_var('LIVECD');
```

Comment:
Suggestion: "We should only reset consoles if the system really rebooted. Otherwise the next select_console will check for a login prompt instead of handling the still logged in system."

This way it's a little more clear that not "we" screw up in looking for the right needle but that it is an expected behavior of `select_console` after `reset_console`

Reply:
ok, I took your comment